dot/dot.go

```diff
@@ -17,13 +17,53 @@
 package dot
 
 import (
-	"github.com/ChainSafe/gossamer/p2p"
-	"github.com/ChainSafe/gossamer/polkadb"
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/rpc"
+	log "github.com/ChainSafe/log15"
 )
 
-// Dot is a container on which services can be registered.
+// Dot is a container for all the components of a node.
 type Dot struct {
-	ServerConfig *p2p.ServiceConfig
-	Server       *p2p.Service      // Currently running P2P networking layer
-	Polkadb      *polkadb.BadgerDB //BadgerDB database
-}
\ No newline at end of file
+	Services *common.ServiceRegistry // Registry of all core services
+	Rpc      *rpc.HttpServer         // HTTP instance for RPC server
+
+	stop chan struct{} // Used to signal node shutdown
+}
+
+// NewDot initializes a Dot with provided components.
+func NewDot(services []common.Service, rpc *rpc.HttpServer) *Dot {
+	d := &Dot{
+		Services: common.NewServiceRegistry(),
+		Rpc:      rpc,
+		stop:     make(chan struct{}),
+	}
+
+	for _, srvc := range services {
+		d.Services.RegisterService(srvc)
+	}
+
+	return d
+}
+
+// Start starts all services. API service is started last.
+func (d *Dot) Start() {
+	log.Debug("Starting core services.")
```

Comment:
`log.Debug` is showing up in the terminal output making the logs rather difficult read

Reply:
Is that to suggest we should turn down the verbosity?

Reply:
Yeah, we need to look into log15 a bit more I think. I don't think `log.Debug` should show up in the terminal output unless explicitly stated in our config upon start up which we do not have in place right now.

Reply:
Looks like we can pass a filter with this https://github.com/ChainSafe/log15/blob/67afb5ed74ec82fd7ac8f49d27c509ac6f991970/handler.go#L185 by using `log.SetHandler()`. Will need to do something in the tests as well as adding CLI flag